File: 2022/2/day2.py
with open("input/day2_input.txt") as f:
    input = f.read().split()

l = list()
d = {
    'A': 0, 'X': 0,
    'B': 1, 'Y': 1,
    'C': 2, 'Z': 2
}

for x in input:
    l.append(d.get(x))

score = 0
for i in range(0, len(l), 2):
    score += l[i + 1] + 1
    if (l[i] + 1) % 3 == l[i + 1]:
        score += 6
    elif l[i] == l[i + 1]:
        score += 3

print("Sol1:", score)

score = 0
for i in range(0, len(l), 2):
    if l[i + 1] == 0:
        score += ((l[i] - 1) % 3) + 1
    elif l[i + 1] == 1:
        score += l[i] + 4
    else:
        score += ((l[i] + 1) % 3) + 7

print("Sol2:", score)

File: 2022/2/day2.c
#include <stdio.h>

int main()
{
    FILE *file = fopen("input/day2_input.txt", "r");
    char line[10];
    int score1 = 0, score2 = 0;
    char opp, you;

    while (fgets(line, 10, file))
    {
        sscanf(line, "%c %c", &opp, &you);
        opp -= 65;
        you -= 88;

        score1 += you + 1;
        if ((opp + 1) % 3 == you % 3)
            score1 += 6;
        else if (opp == you)
            score1 += 3;

        if (you == 0)
            score2 += (opp != 0 ? opp - 1 : opp + 2) + 1;
        else if (you == 1)
            score2 += opp + 4;
        else
            score2 += ((opp + 1) % 3) + 7;
    }

    printf("Sol1: %d\n", score1);
    printf("Sol2: %d\n", score2);
    return 0;
}